certbot/auth_handler.py

```diff
@@ -33,14 +33,18 @@ class AuthHandler(object):
         and values are :class:`acme.messages.AuthorizationResource`
     :ivar list achalls: DV challenges in the form of
         :class:`certbot.achallenges.AnnotatedChallenge`
+    :ivar list pref_challs: sorted user specified preferred challenges
+        in the form of subclasses of :class:`acme.challenges.Challenge`
+        with the most preferred challenge listed first
 
     """
-    def __init__(self, auth, acme, account):
+    def __init__(self, auth, acme, account, pref_challs):
```

Comment:
I presume you were suggesting this because the combination of `pref_challs` and `chall_prefs` is confusing. I agree, though there are several call sites that pass in `pref_challs` from elsewhere in the codebase, and it's probably clearer to have an indication of semantics at those call sites.

Maybe we should rename `chall_prefs`? Or kick this to a refactor ticket?